lib/Core/Executor.h

```diff
@@ -398,6 +398,13 @@ class Executor : public Interpreter {
   ref<klee::ConstantExpr> toConstant(ExecutionState &state, ref<Expr> e, 
                                      const char *purpose);
 
+  /// Evaluate the given expression under each seed, and return the
+  /// first one that results in a constant, if such a seed exist.  Otherwise,
+  /// return the non-constant evaluation of the expression under one of the
+  /// seeds.
+  ref<klee::Expr> getValueFromSeeds(std::vector<SeedInfo> &seeds,
```

Comment:
I would suggest to change the API of this call to use the current state as an argument and return a constant expression.

```
ref<klee::ConstantExpr> getValueFromSeeds(ExecutionState& state, const ref<Expr> &e);
```

As it stands now, all of the calls to`getValueFromSeeds` are guarded by a look-up of the seed vector with a subsequent check if the returned value is a constant expr.
Moving that lookup into the function itself makes that function easier to use.

This would simplify the code around quite a lot.